### PR TITLE
refactor: share project and service fields

### PR DESCRIPTION
**Pull request by byashimov**
- Fixes project regex pattern
- ServceName and Project fields are shared across all types
- Both fields are marked as immutable

### CODE REVIEW

**Comment by byashimov** [comment]
Reviewers, please validate regexps.

File: charts/aiven-operator-crds/templates/aiven.io_cassandras.yaml [hunk 1]
```diff
@@ -139,8 +139,8 @@ spec:
                   maxLength: 128
                   type: string
                 project:
-                  description: Target project.
-                  format: ^[a-zA-Z0-9_-]*$
+                  description: Identifies the project this resource belongs to
+                  format: ^[a-zA-Z0-9_-]+$
```
Comment (Serpentiel) [inline charts/aiven-operator-crds/templates/aiven.io_cassandras.yaml:143]: This can't be an empty string, right?

Comment (byashimov) [reply charts/aiven-operator-crds/templates/aiven.io_cassandras.yaml:143]: Can't imagine why project name would be empty. The field is required, though.

**Review by Serpentiel — APPROVED** [review]
lgtm

**Pull request merged**